### PR TITLE
tidb: Fix panic in driverStmt.Query

### PR DESCRIPTION
**Pull request by shenli**
The prepared statement maybe not query, so the resultset maybe nil.

### CODE REVIEW

File: tidb_test.go [hunk 1]
```diff
@@ -257,6 +257,20 @@ func (s *testMainSuite) TestDriverPrepare(c *C) {
...
 
+// Testcase for delete panic
+func (s *testMainSuite) TestDeletePanic(c *C) {
+	db, err := sql.Open("tidb", "memory://test")
+	defer db.Close()
+	_, err = db.Exec("create table t (c int)")
+	c.Assert(err, IsNil)
+	_, err = db.Exec("insert into t values (1), (2), (3)")
+	c.Assert(err, IsNil)
+	_, err = db.Query("delete from `t` where `c` = ?", 1)
```
Comment (siddontang) [inline tidb_test.go]: Query should returns an empty row for delete.

Comment (shenli) [inline tidb_test.go]: PTAL

**Comment by shenli** [comment]
PTAL

**Comment by siddontang** [comment]
should this can fix #57 ?

File: driver.go [hunk 2]
```diff
@@ -379,6 +388,9 @@ func newdriverRows(rs rset.Recordset) *driverRows {
 // result is inferred from the length of the slice.  If a particular column
 // name isn't known, an empty string should be returned for that entry.
 func (r *driverRows) Columns() []string {
+	if r.rs == nil {
+		return make([]string, 0)
```
Comment (qiuyesuifeng) [inline driver.go]: can return nil here?

**Comment by shenli** [comment]
I am not sure. I run go-xorm tidb test with another error. So I can not reach the panic point the same as lunny. 
I will find it out later.

File: driver.go (hunk 2 above)
Comment (qiuyesuifeng) [inline driver.go]: if return an empty slice, using []string{} is better?

Comment (shenli) [inline driver.go]: I am not sure. I think return empty array is better than nil.

Comment (siddontang) [inline driver.go]: []string{} +1

Comment (shenli) [inline driver.go]: PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**